tests/integration/ceph_base_block_test.go

```diff
@@ -45,6 +46,219 @@ func checkSkipCSITest(t *testing.T, k8sh *utils.K8sHelper) {
+	restorePVCName := "restore-block-pvc"
+	// check snapshot is in ready state
+	ready, err := k8sh.CheckSnapshotISReadyToUse(snapshotName, defaultNamespace, 15)
+	require.NoError(s.T(), err)
+	require.True(s.T(), ready, "make sure snapshot is in ready state")
+	// create restore from snapshot and bind it to app
+	logger.Infof("step 10: restore pvc to a new snapshot")
+	err = helper.BlockClient.CreatePVCRestore(defaultNamespace, restorePVCName, snapshotName, storageClassName, "ReadWriteOnce", pvcSize)
+	require.NoError(s.T(), err)
+	require.True(s.T(), k8sh.WaitUntilPVCIsBound(defaultNamespace, restorePVCName), "Make sure PVC is Bound")
+
+	restorePodName := "restore-pod"
+	logger.Infof("step 11: Bind PVC Restore to application")
+	err = helper.BlockClient.CreatePod(restorePodName, restorePVCName, defaultNamespace, mountPoint, readOnly)
+	assert.NoError(s.T(), err)
+
+	logger.Infof("step 12: check pod is in running state")
+	require.True(s.T(), k8sh.IsPodRunning(restorePodName, defaultNamespace), "make sure pod is in running state")
+	logger.Infof("Storage Mounted successfully")
+
+	// get the checksum of the data and validate it
+	logger.Infof("step 13: check md5sum of both pvc and restore data is same")
+	cmd = fmt.Sprintf("md5sum %s/file.out", mountPoint)
+	resp, err = k8sh.RunCommandInPod(defaultNamespace, restorePodName, cmd)
+	require.NoError(s.T(), err)
+	restorePVCChecksum := strings.Fields(resp)
+	require.Equal(s.T(), len(restorePVCChecksum), 2)
+
+	// compare the checksum value and verify the values are equal
+	assert.Equal(s.T(), restorePVCChecksum[0], pvcChecksum[0])
+	// delete clone PVC and app
+	logger.Infof("step 14: delete restore pod")
+
+	err = k8sh.DeletePod(k8sutil.DefaultNamespace, restorePodName)
+	require.NoError(s.T(), err)
+	logger.Infof("step 15: delete restore pvc")
+
+	err = helper.BlockClient.DeletePVC(defaultNamespace, restorePVCName)
+	assertNoErrorUnlessNotFound(s, err)
+	assert.True(s.T(), k8sh.WaitUntilPVCIsDeleted(defaultNamespace, restorePVCName))
+
+	// delete the snapshot
+	logger.Infof("step 16: delete snapshot")
+
+	err = helper.BlockClient.DeleteSnapshot(snapshotName, pvcName, snapshotClassName, defaultNamespace)
+	require.NoError(s.T(), err)
+	logger.Infof("step 17: delete application pod")
+
+	// delete the parent PVC and app
+	err = k8sh.DeletePod(k8sutil.DefaultNamespace, podName)
+	require.NoError(s.T(), err)
+	logger.Infof("step 18: delete parent pvc")
+
+	err = helper.BlockClient.DeletePVC(defaultNamespace, pvcName)
+	assertNoErrorUnlessNotFound(s, err)
+	assert.True(s.T(), k8sh.WaitUntilPVCIsDeleted(defaultNamespace, pvcName))
+
+	logger.Infof("step 19: delete snapshotclass")
+
+	err = helper.BlockClient.DeleteSnapshotClass(snapshotClassName, snapshotDeletePolicy, namespace)
+	require.NoError(s.T(), err)
+	logger.Infof("step 20: delete snapshot-controller")
+
+	err = k8sh.DeleteSnapshotController()
+	require.NoError(s.T(), err)
+	logger.Infof("step 21: delete snapshot CRD")
```

Comment:
Numbering the steps isn't great because if steps are added or removed, they all have to change numbers. I know some other tests use the step numbers, but I don't find them very useful. As long as the strings logged are searchable, it's fine without step numbers.

Reply:
removed numbering